rules/dispatch.go

```diff
@@ -102,112 +113,69 @@ func (r *DefaultRuleRenderer) hostDispatchChains(
 	}
 
 	if fromOnly {
-		return r.dispatchChains(
+		return r.dispatchEndpointChains(
 			names,
-			epMarkMapper,
 			HostFromEndpointPfx,
 			"",
 			ChainDispatchFromHostEndpoint,
 			"",
 			false,
 			false,
-			false,
 		)
 	}
 
 	if !applyOnForward {
-		return r.dispatchChains(
+		return r.dispatchEndpointChains(
 			names,
-			epMarkMapper,
 			HostFromEndpointPfx,
 			HostToEndpointPfx,
 			ChainDispatchFromHostEndpoint,
 			ChainDispatchToHostEndpoint,
 			false,
 			false,
-			false,
 		)
-
 	}
 
 	return append(
-		r.dispatchChains(
+		r.dispatchEndpointChains(
 			names,
-			epMarkMapper,
 			HostFromEndpointPfx,
 			HostToEndpointPfx,
 			ChainDispatchFromHostEndpoint,
 			ChainDispatchToHostEndpoint,
 			false,
 			false,
-			false,
 		),
-		r.dispatchChains(
+		r.dispatchEndpointChains(
 			names,
-			epMarkMapper,
 			HostFromEndpointForwardPfx,
 			HostToEndpointForwardPfx,
 			ChainDispatchFromHostEndPointForward,
 			ChainDispatchToHostEndpointForward,
 			false,
 			false,
-			false,
 		)...,
 	)
 }
 
-func (r *DefaultRuleRenderer) dispatchChains(
+func (r *DefaultRuleRenderer) dispatchEndpointChains(
 	names []string,
-	epMarkMapper EndpointMarkMapper,
 	fromEndpointPfx,
 	toEndpointPfx,
 	dispatchFromEndpointChainName,
 	dispatchToEndpointChainName string,
 	dropAtEndOfFromChain bool,
 	dropAtEndOfToChain bool,
-	useEndPointMark bool,
 ) []*Chain {
 	// Sort interface names so that rules in the dispatch chain are ordered deterministically.
```

Comment:
Why not move this sort into the `divideEndpointNamesToPrefixTree` function so it can be shared?

Reply:
Good point. Sort can be shared.